src/cpp/dds/domain/DomainParticipant.cpp

```diff
@@ -129,15 +129,7 @@ dds::pub::qos::PublisherQos DomainParticipant::default_publisher_qos() const
 DomainParticipant& DomainParticipant::default_publisher_qos(
         const ::dds::pub::qos::PublisherQos& qos)
 {
-    ReturnCode_t code = this->delegate()->set_default_publisher_qos(qos);
-    if (code == ReturnCode_t::RETCODE_INCONSISTENT_POLICY)
-    {
-        throw dds::core::InconsistentPolicyError("Inconsistent Qos");
-    }
-    else if (code == ReturnCode_t::RETCODE_IMMUTABLE_POLICY)
-    {
-        throw dds::core::ImmutablePolicyError("Immutable Qos");
-    }
+    this->delegate()->set_default_publisher_qos(qos);
```

Comment:
It was OK to check for INCONSISTENT. Also check for UNSUPPORTED.
No need for IMMUTABLE, default_publisher_qos can always be changed